src/blockdata/transaction.rs

```diff
@@ -128,7 +128,18 @@ impl Default for TxOut {
     }
 }
 
-/// A Bitcoin transaction, which describes an authenticated movement of coins
+/// A Bitcoin transaction, which describes an authenticated movement of coins.
+///
+/// If any inputs have nonempty witnesses, the entire transaction is serialized
+/// in the post-BIP141 Segwit format which includes a list of witnesses. If all
+/// inputs have empty witnesses, the transaction is serialized in the pre-BIP143
```

Comment:
You use BIP143 here instead of 141. 143 is not about encoding.

Reply:
I always do that :). fixed.